src/pe/resource.rs

```diff
@@ -0,0 +1,1681 @@
+
+        if let Some(entry) = it {
+            let offset_to_data =
+                match entry.recursive_next_depth(bytes, |e| e.offset_to_data().is_none())? {
+                    Some(next) => match next.offset_to_data() {
+                        Some(offset_to_data) => offset_to_data,
+                        None => return Ok(None),
+                    },
+                    None => return Ok(None),
+                };
+            let mut offset = offset_to_data as usize;
+            let data_entry = bytes.gread_with::<ResourceDataEntry>(&mut offset, scroll::LE)?;
+            let rva = data_entry.offset_to_data as usize;
+            offset = utils::find_offset(rva, sections, file_alignment, opts).ok_or_else(|| {
+                error::Error::Malformed(format!(
+                    "Cannot map ResourceDataEntry rva {:#x} into offset",
+                    rva
+                ))
+            })?;
+
+            if offset + data_entry.size as usize > pe.len() {
+                return Err(error::Error::Malformed(format!(
+                    "offset ({:#x}) and data_entry.size ({:#x}) is greater than pe len {:#x}",
+                    offset,
+                    data_entry.size,
+                    bytes.len()
+                )));
+            }
+            let data = &pe[offset..offset + data_entry.size as usize];
+            let iterator = ResourceStringIterator { data };
+            let strings = iterator.collect::<Result<Vec<_>, _>>()?;
+
+            let fixed_info = match strings
+                .iter()
+                .find(|x| x.key_string() == VS_VERSION_INFO_KEY)
+            {
+                Some(version_info) => Some(version_info.value.pread_with(0, scroll::LE)?),
+                None => None,
+            };
+            let string_info = StringFileInfo::from_resource_string_iterator(iterator);
+
+            Ok(Some(Self {
+                data,
+                fixed_info,
+                string_info,
+            }))
+        } else {
+            Ok(None)
+        }
+    }
+}
+
+/// Represents a manifest data within resource in the PE (Portable Executable) format.
+#[derive(Copy, Clone, Default)]
+pub struct ManifestData<'a> {
+    /// The raw binary data of the manifest
+    pub data: &'a [u8],
+}
+
+impl fmt::Debug for ManifestData<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("ManifestData")
+            .field("value", &format_args!("{:02x?}", self.data))
+            .finish()
+    }
+}
+
+impl<'a> ManifestData<'a> {
+    pub fn parse(
+        pe: &'a [u8],
+        bytes: &'a [u8],
+        it: ResourceEntryIterator<'a>,
+        sections: &[section_table::SectionTable],
+        file_alignment: u32,
+        opts: &options::ParseOptions,
+    ) -> error::Result<Option<Self>> {
+        let it = it.collect::<Result<Vec<_>, _>>()?;
+        let it = it.iter().find(|e| e.id() == Some(RT_MANIFEST));
```

Comment:
ditto here; all of this code looks very similar, with only minor variations; my brain immediately wants to abstract it somehow, but it will likely have very little gain, and likely only add complication :)